### PR TITLE
feat: parse relative domains as links

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

https://app.clickup.com/t/1qe65cv

With this PR, links that contain a domain (valid hostname) without the schema (`http`) will be parsed as links.

Considerations:

Theoretically is possible to have a relative path with a dot `.` if that is the case, it's possible that this new change will consider those paths as links. For example, `(my path)[path.withdot/something]` will be rendered like `http://path.withdot/something.` I don't think this will be a common issue (or an issue at all), but certainly not impossible.

As a reference, the only "local" links that may contain a dot that I was able to think on our projects are the docs, and those are parsed properly (see tests), for example: `docs/core/releases/upgrade/docker/3.0`

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   x ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: src/CommonMark/Extensions/Link/LinkRenderer.php [hunk 1]
```diff
@@ -95,6 +95,31 @@ public function getXmlAttributes(Node $node): array
...
     }
 
+    private function getNodeUrl(Node $node): string
+    {
+        $url = $node->getUrl();
+
+        $path = parse_url($url, PHP_URL_PATH);
+
+        if ($this->pathIsADomain($path)) {
+            return 'http://'.$path;
```
Comment (ItsANameToo) [inline src/CommonMark/Extensions/Link/LinkRenderer.php:105]:
```suggestion
            return 'https://'.$path;
```
wouldn't https be better? 🤔

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: src/CommonMark/Extensions/Link/LinkRenderer.php (hunk 1 above)
Comment (alfonsobries) [reply src/CommonMark/Extensions/Link/LinkRenderer.php:105]: @ItsANameToo I used http because not all sites use https but all use http

Comment (faustbrian) [reply src/CommonMark/Extensions/Link/LinkRenderer.php:105]: If a site doesn't use a secure setup I would say that's on them. By default we should redirect the user to the one with an SSL certificate and if that doesn't work then the user should go to the insecure one on their own.

**Comment by alfonsobries** [comment]
@ItsANameToo ready

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**